src/spikeinterface/core/tests/test_time_handling.py

```diff
@@ -305,7 +311,87 @@ def test_sorting_analyzer_get_durations_no_recording(self, time_vector_recording
 
         assert np.array_equal(sorting_analyzer.get_total_duration(), raw_recording.get_total_duration())
 
-    # Helpers ####
+    @pytest.mark.parametrize("fixture_name", ["time_vector_recording", "t_start_recording"])
+    @pytest.mark.parametrize("shift", [-123.456, 123.456])
+    def test_shift_time_all_segments(self, request, fixture_name, shift):
+        """
+        Shift the times in every segment using the `None` default, then
+        check that every segment of the recording is shifted as expected.
+        """
+        _, times_recording, all_times = self._get_fixture_data(request, fixture_name)
+
+        num_segments, orig_seg_data = self._store_all_times(times_recording)
+
+        times_recording.shift_times(shift)  # use default `segment_index=None`
+
+        for idx in range(num_segments):
+            assert np.allclose(
+                orig_seg_data[idx], times_recording.get_times(segment_index=idx) - shift, rtol=0, atol=1e-8
+            )
+
+    @pytest.mark.parametrize("fixture_name", ["time_vector_recording", "t_start_recording"])
+    @pytest.mark.parametrize("shift", [-123.456, 123.456])
+    def test_shift_times_different_segments(self, request, fixture_name, shift):
+        """
+        Shift each segment separately, and check the shifted segment only
+        is shifted as expected.
+        """
+        _, times_recording, all_times = self._get_fixture_data(request, fixture_name)
+
+        num_segments, orig_seg_data = self._store_all_times(times_recording)
```

Comment:
Maybe you could just do recording.copy?

Reply:
Nice this will allow to remove a lot of boiler plate and make the assert more explicit. However check this out!:

```
        import copy

        _, times_recording, all_times = self._get_fixture_data(request, fixture_name)

        orig_times_recording = copy.deepcopy(times_recording)

        # num_segments, orig_seg_data = self._store_all_times(times_recording)

        times_recording.shift_times(shift)  # use default `segment_index=None`

        for idx in range(num_segments):
            assert np.allclose(
                orig_times_recording.get_times(segment_index=idx),
                times_recording.get_times(segment_index=idx) - shift,
                rtol=0, atol=1e-8
            )
```
This fails,  `orig_times_recording.get_times(segment_index=idx)` gives 
```
# array([0.00000000e+00, 3.33333333e-05, 6.66666667e-05, ...,
 #      9.99990000e+00, 9.99993333e+00, 9.99996667e+00])
```

It seems the time vector info is not being propagated in the copy, though it seemed to work okay for the save/ load 🤔 will have to come back to this, do you have ideas why this might be happening?

Reply:
Mmm yeah, I think that that's probably by design. I think cloning uses to do_dict mechanism and that coupled with saving to disk and serializing. Because timestamps can be large we probably don't want that there.

The solution is to change copy (which should be a in memory thing) to copy the time vector but I think that should be done in another PR.

Reply:
A little out of scope, but do you remember when we were looking up copy vs clone at Edinburgh. I forget what it said (but basically one is deep and one is suppose to be shallow). It would probably be worth us making clear in our docstrings if our copy's are shallow or not (also for another PR).

Reply:
> I'm a little confused sorry, what is `clone` vs `copy`? BTW I forgot to say based on the suggested command, I got an error:
> 
> ```
> (Pdb) times_recording.copy()
> *** AttributeError: 'NoiseGeneratorRecording' object has no attribute 'copy'
> ```
> 
> should this have worked?

No I don't think this would work. I guess @h-mayorquin was suggesting to do a:
```
from copy import copy

copy(times_recording)
```

Reply:
> I'm a little confused sorry, what is clone vs copy?

Some projects use `copy(deep=True/False)` to indicate whether to just do metadata vs actual array data whereas other projects use `clone` vs `copy` to make the deep vs shallow distinction (again deep being everything and shallow just being metadata). Because there is no requirement for clone and copy being shallow or deep it would probably be better for us as a project to either make it clear in the docstring or for us to use the `clone/copy` names to make clear which functions are deep and which are shallow. Is that clearer? Again not crucial to this PR, but related to the fact that the array is not being copied (ie our copy is shallow).